Source/utils/file_util.cpp

```diff
@@ -73,6 +73,27 @@ bool FileExists(const char *path)
 #endif
 }
 
+bool IsWriteable(const char *path)
+{
+#if defined(_WIN64) || defined(_WIN32)
+	const auto pathUtf16 = ToWideChar(path);
+	if (pathUtf16 == nullptr) {
+		LogError("UTF-8 -> UTF-16 conversion error code {}", ::GetLastError());
+		return false;
+	}
+	uint32_t attr = ::GetFileAttributesW(&pathUtf16[0]);
+	if (attr != INVALID_FILE_ATTRIBUTES && (attr & FILE_ATTRIBUTE_READONLY) != 0) {
+		return false;
+	}
+	return true;
+#elif _POSIX_C_SOURCE >= 200112L || defined(_BSD_SOURCE) || defined(__APPLE__)
+	return ::access(path, W_OK) == 0;
+#else
+	struct stat stats;
+	return stat(path, &stats) == 0 && (stats.st_mode & W_OK) != 0;

```

Comment:
From what I can tell, `W_OK` is an `::access` flag, not a `stat` flag?

Anyway, I think this `#else` clause is no longer necessary and can be removed. `access` is available whenever `stat` is available, they're both POSIX https://pubs.opengroup.org/onlinepubs/9699919799/functions/access.html

Reply:
Removed